### PR TITLE
Migrate `database` to Kotlin

### PR DESCRIPTION
**Pull request by wiryadev**
Migrate all files inside `database` package to Kotlin

### CODE REVIEW

File: app/src/main/java/com/quran/labs/androidquran/database/DatabaseUtils.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+object DatabaseUtils {
+  @JvmStatic
+  fun closeCursor(cursor: Cursor?) {
+    if (cursor != null) {
+      try {
+        cursor.close()
+      } catch (e: Exception) {
+        // no op
+      }
+    }
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/database/DatabaseUtils.kt:15]:
```kotlin
try {
  cursor?.close()
} catch (e: Exception) {
   // no op
}
```

File: app/src/main/java/com/quran/labs/androidquran/database/SuraTimingDatabaseHandler.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        ),
+        TimingsTable.COL_SURA + "=" + sura,
+        null, null, null, TimingsTable.COL_AYAH + " ASC"
+      )
+    } catch (e: Exception) {
+      null
+    }
+  }
+
+  val version: Int
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/database/SuraTimingDatabaseHandler.kt:96]: let's change to a function instead

File: app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import android.util.SparseArray
+import androidx.annotation.WorkerThread
+import com.quran.labs.androidquran.common.LocalTranslation
+import com.quran.labs.androidquran.dao.translation.TranslationItem
+import com.quran.labs.androidquran.database.TranslationsDBHelper.TranslationsTable
+import com.quran.labs.androidquran.util.QuranFileUtils
+import timber.log.Timber
+import java.util.*
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt:14]: let's avoid star imports

File: app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    for (item in translations) {
+      result.put(item.id, item)
+    }
+    return result
+  }
+
+  // intentional, since cachedTranslations can be replaced by another thread, causing the check
+  // to be true, but the cached object returned to be null (or to change).
+  @get:WorkerThread
+  val translations: List<LocalTranslation>
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt:43]: let's make it a function instead

**Review by ahmedre — COMMENTED** [review]
jazakumAllah khairan - some minor comments - but please let's try to keep smaller PRs

File: app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt (hunk 4 above)
Comment (wiryadev) [reply app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt:43]: > minor

yes brother, will make sure of that in the future

**Comment by ahmedre** [comment]
do you want me to merge this and you want to add these changes in a clean PR?

File: app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt (hunk 4 above)
Comment (wiryadev) [reply app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt]: i have push this change, and it needs changes of variable in this l[ine](https://github.com/wiryadev/quran_android/blob/bb9cb39c07d4ff0526a598c6051eb2b58236cc1e/app/src/main/java/com/quran/labs/androidquran/presenter/translation/BaseTranslationPresenter.kt#L180)

from
`Single.fromCallable<List<LocalTranslation>> { translationsAdapter.translations }`

to
`Single.fromCallable<List<LocalTranslation>> { translationsAdapter.getTranslations()}`

**Comment by wiryadev** [comment]
> do you want me to merge this and you want to add these changes in a clean PR?

i'll just add those changes into this PR

**Comment by ahmedre** [comment]
jazakumAllah khairan.

**Pull request merged**